sdk-core/src/main/java/care/data4life/sdk/RecordService.kt

```diff
@@ -610,24 +612,24 @@ class RecordService(
 
         builder.setDataKey(dataKey)
 
-        if (record.encryptedAttachmentsKey != null) {
+        if (record.encryptedAttachmentsKey is EncryptedKey) {
```

Comment:
if you use a value the compiler is able to check for not null and you don't need force !!

```
        val attachmentKey = record.encryptedAttachmentsKey
        if (attachmentKey is EncryptedKey) {
            builder.setAttachmentKey(
                    cryptoService.symDecryptSymmetricKey(
                            commonKey,
                            attachmentKey
                    ).blockingGet()
            )
        }
```

Reply:
done